src/particles/elements/NonlinearLens.H

```diff
@@ -0,0 +1,123 @@
+/* Copyright 2022 Chad Mitchell, Axel Huebl
+ *
+ * This file is part of ImpactX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_NONLINEARLENS_H
+#define IMPACTX_NONLINEARLENS_H
+
+#include "particles/ImpactXParticleContainer.H"
+
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+#include <AMReX_GpuComplex.H>
+
+#include <cmath>
+
+namespace impactx
+{
+    struct NonlinearLens
+    {
+        using PType = ImpactXParticleContainer::ParticleType;
+
+        /** A thin lens associated with a single short segment of the
```

Comment:
This is fantastic!
Let me just split it in "short title" + "long description" with a single empty newline after the title.
```suggestion
        /** Single short segment of the nonlinear magnetic insert element
         *
         *  A thin lens associated with a single short segment of the
```